generator/src/test/java/com/scottlogic/deg/generator/walker/reductive/ReductiveStateTests.java

```diff
@@ -14,64 +14,44 @@
     @Test
     void allFieldsAreFixed_lastFixedFieldNotNullAndFixedFieldsSizePlusOneEqualsFieldsSize_returnsTrue() {
         HashMap<Field, FixedField> fixedFields = new HashMap<>();
-        fixedFields.put(
-            new Field("Second Field"),
-            new FixedField(new Field("Second Field"), null, null, null)
-        );
-        FixedField lastFixedField = new FixedField(
-            new Field("First Field"),
-            null,
-            null,
-            null
-        );
+        fixedFields.put(new Field("Second Field"),
+            new FixedField(new Field("Second Field"), null, null, null));
+        FixedField lastFixedField = new FixedField(new Field("First Field"), null, null, null);
         ReductiveState reductiveState = getReductiveState(
             Arrays.asList("First Field", "Second Field"),
             fixedFields
-        ).with(lastFixedField);
+        ).with(lastFixedField)
+            .with(new FixedField(null, null,null, null));
```

Comment:
It feels 'weird'/'illegal' to have a `FixedField` with no `Field`, i know this is tests but i'm struggling to understand if this is a mistake or required.

i.e. "First Field" and "Second Field" have been fixed on lines 17 and 23, which field is being fixed here?

Maybe you should be calling `whereCurrentFieldIsFixed()`?